### PR TITLE
[WIP] nodejs chart endpoint

### PR DESCRIPTION
**Pull request by chejennifer**
endpoint: /nodejs/chart-info
params:
- place: the dcid of the main place/parent place if using enclosedPlaceType
- enclosedPlaceType: the type of place
- config: tile config of the tile to get the chart info for as a json object
- svSpec: list of stat var specs to show in the chart as a json list

response is a json object:

```
{
  // List of sources of the data in the chart
  srcs: { name: string; url: string }[];
  // The title of the tile
  title: string;
  // The type of the tile
  type: string;
  // List of legend labels
  legend?: string[];
  // The data for the chart in the tile as a csv string
  data_csv: string;
  // The svg for the chart in the tile as an xml string
  svg: string;
}
```

e.g., http://localhost:3030/nodejs/chart-info?config={"barTileSpec":{"maxPlaces":15,"maxVariables":15,"sort":"DESCENDING"},"comparisonPlaces":["geoId/06061"],"title":"Prevalence%20of%20Health%20Conditions%20in%20Placer%20County%20(${date})","type":"BAR"}&place=geoId/06061&svSpec=[{"name":"Arthritis","statVar":"Percent_Person_WithArthritis","unit":"%25"},{"name":"Asthma","statVar":"Percent_Person_WithAsthma","unit":"%25"},{"name":"Cancer%20(excluding%20skin%20cancer)","statVar":"Percent_Person_WithCancerExcludingSkinCancer","unit":"%25"},{"name":"Chronic%20Kidney%20Disease","statVar":"Percent_Person_WithChronicKidneyDisease","unit":"%25"},{"name":"Chronic%20Obstructive%20Pulmonary%20Disease","statVar":"Percent_Person_WithChronicObstructivePulmonaryDisease","unit":"%25"},{"name":"Coronary%20Heart%20Disease","statVar":"Percent_Person_WithCoronaryHeartDisease","unit":"%25"},{"name":"Diabetes","statVar":"Percent_Person_WithDiabetes","unit":"%25"},{"name":"High%20Blood%20Pressure","statVar":"Percent_Person_WithHighBloodPressure","unit":"%25"},{"name":"High%20Cholesterol","statVar":"Percent_Person_WithHighCholesterol","unit":"%25"},{"name":"Mental%20Health%20Issues","statVar":"Percent_Person_WithMentalHealthNotGood","unit":"%25"},{"name":"Physical%20Health%20Issues","statVar":"Percent_Person_WithPhysicalHealthNotGood","unit":"%25"},{"name":"Stroke","statVar":"Percent_Person_WithStroke","unit":"%25"}]

### CODE REVIEW

**Review by pradh — APPROVED** [review]
Nice!

File: static/src/server.ts [hunk 1]
```diff
@@ -538,6 +597,36 @@ app.get("/nodejs/chart", (req: Request, res: Response) => {
...
+  res.setHeader("Content-Type", "application/json");
+  const namedTypedPlace = { dcid: place, name: place, types: []}
+  getTileResult("TEST", namedTypedPlace, enclosedPlaceType, svSpec, tileConfig)
+  .then((tileResult) => {
+    /*
+    const container = document.createElement("div");
+    const img = document.createElement("img");
+    img.src = tileResult.svg;
+    container.appendChild(img);
+  res.setHeader("Content-Type", "text/html").send(container.outerHTML);*/
```
Comment (pradh) [inline static/src/server.ts:620]: nit: delete if not needed

**Review by pradh — APPROVED** [review]

**Comment by pradh** [comment]
To confirm, the tile config does not need the list of sv-spec keys?  https://github.com/datacommonsorg/website/blob/29343f03a5a9fc1fd67fc291408c1c9037f936c9/server/config/subject_page.proto#L275

**Comment by chejennifer** [comment]
> To confirm, the tile config does not need the list of sv-spec keys?
> 
> https://github.com/datacommonsorg/website/blob/29343f03a5a9fc1fd67fc291408c1c9037f936c9/server/config/subject_page.proto#L275

correct, don't need it because we specify the stat vars using svSpec param

**Pull request merged**